### PR TITLE
Improve message logging

### PR DESCRIPTION
**Pull request by jzzheng22**
# Summary
Adds `String()` method so logging can print string type instead of an int.

### CODE REVIEW

**Review by hussain2603 — APPROVED** [review]
LGTM, thanks Jason :)

**Review by udai-arneja — DISMISSED** [review]
LGTM

**Review by udai-arneja — DISMISSED** [review]
🦾

File: pkg/messages/baseMessage_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
+package messages
+
```
Comment (woonmoon) [inline pkg/messages/baseMessage_test.go:2]: Should we create a `test` package?

**Review by woonmoon — COMMENTED** [review]
LGTM? Unsure if testing belongs in the messages package but otherwise LGTM

File: pkg/messages/baseMessage_test.go (hunk 1 above)
Comment (JaafarRammal) [reply pkg/messages/baseMessage_test.go:2]: I think we should be testing those components in a more connected way? like send a message and then receive it and then check it's been received correctly / to the right agent?

cc @jzzheng22

File: pkg/messages/baseMessage.go [hunk 2]
```diff
@@ -23,6 +23,41 @@ const (
 	Response
 )
 
+func (m MessageType) String() string {
```
Comment (JaafarRammal) [inline pkg/messages/baseMessage.go:26]: Can string be a function implemented in each message instead?

**Review by JaafarRammal — CHANGES_REQUESTED** [review]
I think "String()" should be implemented under each message's implementation instead of a case in base. And every time a new message type is added, the String implementation is added too

File: pkg/messages/baseMessage_test.go (hunk 1 above)
Comment (JaafarRammal) [reply pkg/messages/baseMessage_test.go:2]: also @woonmoon I think in go that's how tests are done (you add a _test file in the same package), but I might be wrong

Comment (jzzheng22) [reply pkg/messages/baseMessage_test.go:2]: Ideally we would test the message passing properly... but I'm just getting the easy one here for now. Can create an issue for longer testing... but also only two weeks left.

**Comment by JaafarRammal** [comment]
> I think "String()" should be implemented under each message's implementation instead of a case in base. And every time a new message type is added, the String implementation is added too

On 2nd thoughts, that would mean we'd use `msg.String()` instead of `msg.MessageType().String()`, and the first is confusing since it could mean the message content. I believe this is the best way since:
- Enum key string needs the case
- Go doesn't allow function overload to have a String and INT version of the same function name for each implementation

**Comment by JaafarRammal** [comment]
Pipeline goes brr

File: Makefile [hunk 3]
```diff
@@ -17,15 +17,18 @@ test: dep ## Run unittests
 
 test-coverage: dep ## Run tests with coverage
 	@go test -coverprofile ./bin/cover.out -covermode=atomic ./...
-	@cat ./bin/cover.out
 
 build: dep ## Build the binary file
+	@go install golang.org/x/tools/cmd/stringer
```
Comment (JaafarRammal) [inline Makefile:22]: Can these be removed now 👀

**Review by JaafarRammal — APPROVED** [review]

**Pull request merged**